test/decimal_test.exs

```diff
@@ -747,6 +747,21 @@ defmodule DecimalTest do
     end)
   end
 
+  test "integer?/1" do
+    refute Decimal.integer?(~d"0.1")
+    refute Decimal.integer?(~d"0.10")
+    refute Decimal.integer?(~d"0.1000")
+    refute Decimal.integer?(~d"0.1234")
+    refute Decimal.integer?(~d"-0.1234")
+    refute Decimal.integer?(~d"1e-100")
+
+    assert Decimal.integer?(~d"1.0000")
+    assert Decimal.integer?(~d"1")
+    assert Decimal.integer?(~d"-1")
+    assert Decimal.integer?(%Decimal{coef: 100, exp: -2})
+    assert Decimal.integer?(~d"1e100")
+  end
```

Comment:
we need to add tests for Inf and NaN. I'm not sure what the return value should be for these, I guess we should raise and document that appropriately?

Reply:
I agree I'm not good at math, then I don't know what's the proper result for a not a number or infinite.

I'm inclined to raising an error, something like:

```
** (ArgumentError) can't determine if Inf is an integer. Inf and NaN aren't allowed in integer checks.
```

Reply:
The Ruby version returns `false` if a number is infinite or NaN

Reply:
In java, we can use `intValueExact`:
https://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html#intValueExact()

If the decimal isn't an integer, it throws an exception.

Reply:
regarding error message, I think we could do something similar to:

```
iex(1)> Decimal.compare(1, "nan")
** (Decimal.Error) invalid_operation: operation on NaN
    (decimal 2.0.0-rc.0) lib/decimal.ex:310: Decimal.compare/2
```

wdyt?

regarding Ruby, I believe this is "the" decimal: https://ruby-doc.org/stdlib-2.7.1/libdoc/bigdecimal/rdoc/BigDecimal.html; the other one you mentioned seems to be https://rubygems.org/gems/ruby-decimal which doesn't seem super popular given it's been around for a long time (and that there is something in stdlib)

Reply:
Sorry to switch in here - from the mathematical point of view my suggestion:

NaN - NotANumber should always raise an exception like @wojtekmach suggested
Inf - is a valid Decimal number -> no error -> simply return false

Reply:
Done!

Reply:
Thanks!

I'm sorry - my last comment was the mathematical point of view - so "no operation with NaN" means real operations like add etc.. but positiv?(NaN) => false. So if you look into the code you can find other kind of boolean functions. 3 things: 1. I was wrong the result of NaN! should be false 2. you wrote valid code but please use the code style like the other boolean functions for uniformity reason 3. no implicit conversion for boolean functions (use Decimal.new/1):

What do you think about the following.

existing boolean function positive?/1 :

```elxir
  @spec positive?(t) :: boolean
  def positive?(%Decimal{coef: :NaN}), do: false
  def positive?(%Decimal{coef: 0}), do: false
...
```

a short shot (please have a look - I didn't checked it much - please make tests!) 

```elxir
  def integer?(%Decimal{coef: :NaN} = num), false
  def integer?(%Decimal{coef: :inf}), do: false
  def integer?(%Decimal{} = num) do
    normalized = normalize(num)
    normalized.exp >= 0 or number_after_dot(normalized) == 0
  end

 ```

Thank you and sorry for confusion.

Reply:
No worries! I'll update!

Reply:
Done 👍